cunumeric/module.py

```diff
@@ -6227,3 +6227,740 @@ def bincount(
+        method 4 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 0
+        * beta = 1
+    hazen:
+        method 5 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 1/2
+        * beta = 1/2
+    weibull:
+        method 6 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 0
+        * beta = 0
+    linear:
+        method 7 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 1
+        * beta = 1
+    median_unbiased:
+        method 8 of H&F [1]_.
+        This method is probably the best method if the sample
+        distribution function is unknown (see reference).
+        This method gives continuous results using:
+        * alpha = 1/3
+        * beta = 1/3
+    normal_unbiased:
+        method 9 of H&F [1]_.
+        This method is probably the best method if the sample
+        distribution function is known to be normal.
+        This method gives continuous results using:
+        * alpha = 3/8
+        * beta = 3/8
+    lower:
+        NumPy method kept for backwards compatibility.
+        Takes ``i`` as the interpolation point.
+    higher:
+        NumPy method kept for backwards compatibility.
+        Takes ``j`` as the interpolation point.
+    nearest:
+        NumPy method kept for backwards compatibility.
+        Takes ``i`` or ``j``, whichever is nearest.
+    midpoint:
+        NumPy method kept for backwards compatibility.
+        Uses ``(i + j) / 2``.
+    References
+    ----------
+    .. [1] R. J. Hyndman and Y. Fan,
+       "Sample quantiles in statistical packages,"
+       The American Statistician, 50(4), pp. 361-365, 1996
+    """
+
+    from .logic import isscalar
+
+    dict_methods = {
+        "inverted_cdf": inverted_cdf,
+        "averaged_inverted_cdf": averaged_inverted_cdf,
+        "closest_observation": closest_observation,
+        "interpolated_inverted_cdf": interpolated_inverted_cdf,
+        "hazen": hazen,
+        "weibull": weibull,
+        "linear": linear,
+        "median_unbiased": median_unbiased,
+        "normal_unbiased": normal_unbiased,
+        "lower": lower,
+        "higher": higher,
+        "midpoint": midpoint,
+        "nearest": nearest,
+    }
+
+    axes_set = []
+    original_shape = a.shape
+
+    if (axis is not None) and (not isscalar(axis)):
+        if len(axis) == 1:
+            real_axis = axis[0]
+            a_rr = a
+            axes_set = [real_axis]
+        else:
+            (real_axis, a_rr) = reshuffle_reshape(a, axis)
+            # What happens with multiple axes and overwrite_input = True ?
+            # It seems overwrite_input is reset to False;
+            overwrite_input = False
+            axes_set = axis
+    else:
+        real_axis = axis
+        a_rr = a
+        axes_set = [real_axis]
+
+    input_is_scalar = isscalar(q)
+
+    if input_is_scalar:
+        q_arr = array([q])
```

Comment:
Creating a cunumeric (i.e. accelerated, distributed) array here is overkill. Since we need to iterate over the elements of this array inside the python program, I think it will be better to just replace this block with `q_arr = np.asarray(q)`. This way we get a NumPy array (i.e. a python-level object), that we can access cheaply in python. Conversely, if we want to access the elements of a full-fledged cunumeric array one-by-one, we would need to negotiate with Legion to create a local instance of the backing region, pulling data from wherever the array has been sharded to. This works, of course, but is not efficient for tiny arrays.

Reply:
Done (modified to use `asarray(q)`).

Reply:
Note that `asarray(q)` calls cunumeric's version of `asarray`, which still creates a cunumeric array. You want to use `np.array(q)` explicitly, because in this file we `import` the vanilla numpy library as `np`:
```
import numpy as np
```

Reply:
Done...I think. I assumed you meant `np.asarray(q)` as in the post just before the one above.

Reply:
`np.array(q)` will always create a copy, even if `q` is already a NumPy array, so in this case `np.asarray` is slightly preferable.